core/db/migrate/20230607100229_import_existing_permission_sets.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class ImportExistingPermissionSets < ActiveRecord::Migration[7.0]
```

Comment:
Did you consider adding this code as a rake task instead? Not everyone will need it and if we can save a migration I think it's generally a good principle to follow. 

Also because they still need to remove the subclasses manually after the migration is run otherwise the permissions will be loaded twice (?).

We could make sure people run it with some check at rails startup, informing that they can and should run the command (in all environments) and after that, redeploy their code with the subclasses removed. 

Even better if we can make subclasses and `Spree::Role` instances live together, giving priority to the latter so that they don't have to rush deleting the subclasses. 

Does it make sense?

Reply:
Thank you for your feedback @kennyadsl!

> Does it make sense?

Yes! I agree with your points and believe your suggestions will make the transition smoother.

So to recap your suggestions:
- Moving the migration to a subsequent PR
- Rake task instead of migration
- Startup check with a message informing devs about the rake task (+ subclasses to remove)
- Coexistence of subclasses and `Spree::Role` Instances

>Also because they still need to remove the subclasses manually after the migration is run otherwise the permissions will be loaded twice (?).

I'll check but probably yes 🤔